src/main/java/io/numaproj/pulsar/producer/PulsarConfig.java

```diff
@@ -31,8 +41,24 @@ public PulsarClient pulsarClient(PulsarClientProperties pulsarClientProperties)
     @Bean
     public Producer<byte[]> pulsarProducer(PulsarClient pulsarClient, PulsarProducerProperties pulsarProducerProperties)
             throws Exception {
+        String podName = env.getProperty("NUMAFLOW_POD", "pod-" + UUID.randomUUID());
+
+        if (podName == null || podName.isBlank()) {
+            podName = "pod-" + UUID.randomUUID();
+            log.warn("NUMAFLOW_POD environment variable not found. Generating fallback Producer name: {}", podName);
+        }
+
+        // Always override the user-specified producerName with the pod name
+        Map<String, Object> producerConfig = pulsarProducerProperties.getProducerConfig();
+        if (producerConfig.containsKey("producerName")) {
```

Comment:
make "producerName" a local variable instead of hardcoded string. that way we have a single place to put the string.